src/layouts/rails-component-layout.tsx

```diff
@@ -1,33 +1,260 @@
+        parameters {
+          default
+          description
+          name
+          type
+        }
+      }
+
+      previews {
+        name
+        preview_path
+        inspect_path
+      }
+
+      subcomponents {
+        name: component
+        railsId: fully_qualified_name
+        description
+        status
+        a11y_reviewed
+        short_name
+
+        props: parameters {
+          name
+          type
+          description
+          default
+        }
+
+        slots {
+          description
+          name
+          parameters {
+            default
+            description
+            name
+            type
+          }
+        }
+
+        methods {
+          name
+          description
+          parameters {
+            default
+            description
+            name
+            type
+          }
+        }
+
+        previews {
+          name
+          preview_path
+          inspect_path
         }
       }
     }
   }
 `
 
+const baseUrl = "https://primer.style/view-components"
+
+function RailsComponentArguments({props, parentRailsId}) {
+  if (props.length > 0) {
+    return (<>
+      <H2>Arguments</H2>
+      <PropsTable props={props} parentRailsId={parentRailsId}/>
+    </>)
+  } else {
+    return <></>
+  }
+}
+
+function RailsComponentSlots({slots, parentRailsId}) {
+  if (slots.length > 0) {
+    return(<>
+      <H2>Slots</H2>
+      {slots.map( (slot) => {
+        return(<>
+          <H3><InlineCode>{slot.name}</InlineCode></H3>
+          {/* @ts-ignore */}
+          <RailsMarkdown text={slot.description} parentRailsId={parentRailsId}/>
+          <PropsTable props={slot.parameters} parentRailsId={parentRailsId}/>
+        </>)
+      })}
+    </>)
+  } else {
+    return <></>
+  }
+}
+
+function RailsComponentMethods({methods, parentRailsId}) {
+  if (methods.length > 0) {
+    return(<>
+      <H2>Methods</H2>
+      {methods.map( (method) => {
+        return(<>
+          <H3><InlineCode>{method.name}</InlineCode></H3>
+          {/* @ts-ignore */}
+          <RailsMarkdown text={method.description} parentRailsId={parentRailsId}/>
+          <PropsTable props={method.parameters} parentRailsId={parentRailsId}/>
+        </>)
+      })}
+    </>)
+  } else {
+    return <></>
+  }
+}
+
+function RailsComponentPreviews({previews, showPreviews}) {
+  if (showPreviews && previews.length > 0) {
+    return(<>
+      <H2>Examples</H2>
+      <LookbookEmbed height={300} previews={previews}/>
+    </>)
+  } else {
+    return <></>
+  }
+}
+
+function RailsComponent({data, showPreviews, parentRailsId}) {
+  const {props, slots, methods, previews} = data
+
+  return(<>
+    <RailsComponentArguments props={props} parentRailsId={parentRailsId}/>
+    <RailsComponentSlots slots={slots} parentRailsId={parentRailsId}/>
+    <RailsComponentMethods methods={methods} parentRailsId={parentRailsId}/>
+    <RailsComponentPreviews previews={previews} showPreviews={showPreviews}/>
```

Comment:
Is there a reason you put the examples under the API reference? In the React pages, we have examples before props.

Reply:
I think I was following the order we have things in our legacy Gatsby docsite, but I'd be more than happy to swap things around 👍